activerecord/lib/active_record/encryption/configurable.rb

```diff
@@ -51,7 +51,11 @@ def encrypted_attribute_was_declared(klass, name) # :nodoc:
         def install_auto_filtered_parameters_hook(application) # :nodoc:
           ActiveRecord::Encryption.on_encrypted_attribute_declared do |klass, encrypted_attribute_name|
             filter_parameter = [("#{klass.model_name.element}" if klass.name), encrypted_attribute_name.to_s].compact.join(".")
-            application.config.filter_parameters << filter_parameter unless excluded_from_filter_parameters?(filter_parameter)
+
+            unless excluded_from_filter_parameters?(filter_parameter)
+              application.config.filter_parameters << filter_parameter
+              klass.filter_attributes << encrypted_attribute_name.to_s
```

Comment:
I think we may want to concatenate arrays instead of pushing to avoid modifying `ActiveRecord::Base.filter_attributes` array. Because, as you pointed out, parent array as return as a default value
https://github.com/rails/rails/blob/190291dc45ddaaaf60a2ea58c2015ceb465719d6/activerecord/lib/active_record/core.rb#L371

```suggestion
              klass.filter_attributes += [encrypted_attribute_name.to_s]
```

I think we may want to add a test to make sure that `ActiveRecord::Base.filter_attributes` is not being modified. Currently `self.filter_attributes = []` hides this behaviour

Reply:
Great call here @nvasilevski!! I've just pushed an update that hopefully resolves what you pointed out.